web/src/auth/LoginPage.js

```diff
@@ -60,11 +60,16 @@ class LoginPage extends React.Component {
       validEmailOrPhone: false,
       validEmail: false,
       validPhone: false,
+      owner:null,
     };
+    if(this.state.type=="cas" &&props.match?.params.casApplicationName !== undefined){
+        this.state.owner=props.match?.params.owner
+        this.state.applicationName=props.match?.params.casApplicationName
+    }
   }
 
   UNSAFE_componentWillMount() {
-    if (this.state.type === "login") {
+    if (this.state.type === "login" || this.state.type === "cas") {
```

Comment:
Why "cas" is comparable with "login", it's not looking right.

Reply:
Because they share the same logic. Under both condition the name of organization and application are known and need to call get-application api